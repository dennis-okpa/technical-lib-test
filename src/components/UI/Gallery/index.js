import React from 'react'
import { Grid } from './styles'
import { useFetch } from '../../../utils/hooks/fetch'
import Row from '../Row/index'
import { getGridData } from '../../../utils/translations/grid'

function Gallery() {
  const data = useFetch()

  if (!data) return null

  const gridData = getGridData(data.rows)

  return (
    <Grid>
      {gridData.map((rowData, index) => (
        <Row key={`gallery-grid-row-${index}`} data={rowData} />
      ))}
    </Grid>
  )
}
export default Gallery
