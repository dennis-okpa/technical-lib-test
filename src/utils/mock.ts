export const mockLibraryData = {
  duration: 33.2277043024078,
  email: 'Ron.Hamill@gmail.com',
  firstName: 'Brooklyn',
  imageUrl:
    'https://images.pexels.com/photos/8422301/pexels-photo-8422301.jpeg',
  lastName: 'Flatley',
  likes: 88,
  thumbnailUrl:
    'https://images.pexels.com/photos/6994926/pexels-photo-6994926.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280',
}
