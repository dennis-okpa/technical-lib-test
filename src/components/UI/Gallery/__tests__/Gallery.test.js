import Gallery from '../index'
import React from 'react'
import { render } from '@testing-library/react'
import { mockLibraryData } from '../../../../utils/mock'

jest.mock('../../../../utils/hooks/fetch', () => ({
  useFetch: () => [
    {
      rows: [mockLibraryData],
    },
    jest.fn(),
  ],
}))

describe('Gallery', () => {
  it('should render correctly', () => {
    const { container } = render(<Gallery />)
    expect(container).toMatchSnapshot()
  })
})
