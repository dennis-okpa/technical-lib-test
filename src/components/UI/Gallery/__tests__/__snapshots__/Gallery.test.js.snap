// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL"
  >
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component "
        style="height: auto; overflow: auto;"
      >
        <h4>
          Loading...
        </h4>
      </div>
    </div>
  </div>
</div>
`;
